pkg/kconfig/kconfig.go

```diff
@@ -63,22 +69,51 @@ func IsBPFEnabled() (bool, error) {
 	return false, fmt.Errorf("kernel config not found")
 }
 
+func checkBPFOption(kernelConfig map[string]string, option string) (bool, error) {
+	value, found := kernelConfig[option]
+	if !found {
+		return false, fmt.Errorf("kernel config required for eBPF not found, Config Option:%s", option)
+	}
+
+	if value != "y" && value != "m" {
+		return false, fmt.Errorf("kernel config required for eBPF is disabled, Config Option:%s", option)
+	}
+	return true, nil
+}
+
 func checkBPFOptions(configFile string) (bool, error) {
 	kernelConfig, err := getConfig(configFile)
 	if err != nil {
 		return false, err
 	}
 
 	for _, option := range ebpfOptions {
-		value, found := kernelConfig[option]
-		if !found {
-			return false, fmt.Errorf("kernel config required for eBPF not found, Config Option:%s", option)
-		}
-
-		if value != "y" && value != "m" {
-			return false, fmt.Errorf("kernel config required for eBPF is disabled, Config Option:%s", option)
+		// Check for the 'primary' ebpf kernel option
+		found, err := checkBPFOption(kernelConfig, option.name)
+		if err != nil {
+			// If there are alternative kernel options specified, iterate over them
+			if len(option.alternatives) > 0 {
+				// Iterate over the list of alternative options and check them sequentially
+				var altFound bool
+				for _, alt := range option.alternatives {
+					altFound, _ = checkBPFOption(kernelConfig, alt)
+					// We only need one of the alternatives specified, so stop searching if found
+					if altFound {
+						break
+					}
+				}
+				if !altFound {
+					// If we reach this point, we were unable to verify the presence of any alternatives
+					alts := strings.Join(option.alternatives, ", ")
+					return false, fmt.Errorf("%w; alternatives checked:%s", err, alts)
+				}
+			} else {
```

Comment:
I would instead negate the condition, return early with this case, and eliminate the need for `else`.
It would make it easier to read. We're already in the 3rd level. (for, if, if). 

Similarly, we can first check for `found` and return even from the function.

Reply:
Ah yeh, fair enough. Will amend shortly.